src/Microsoft.AspNet.Mvc.ModelBinding/Binders/ComplexModelDtoModelBinder.cs

```diff
@@ -19,8 +19,10 @@ public sealed class ComplexModelDtoModelBinder : IModelBinder
                 var dto = (ComplexModelDto)bindingContext.Model;
                 foreach (var propertyMetadata in dto.PropertyMetadata)
                 {
-                    var propertyModelName = ModelBindingHelper.CreatePropertyModelName(bindingContext.ModelName,
-                                                                                       propertyMetadata.PropertyName);
+                    var propertyModelName =
+                        ModelBindingHelper.CreatePropertyModelName(
+                            bindingContext.ModelName,
+                            propertyMetadata.BinderModelName ?? propertyMetadata.PropertyName);
```

Comment:
isn't fromheader doing the opposite? Would be great if `IModelNameProvider` had consistent behavior.